File: clients/ruby/model/jump_state.rb
class JumpState
    attr_accessor :can_jump
    attr_accessor :speed
    attr_accessor :max_time
    attr_accessor :can_cancel
    def initialize(can_jump, speed, max_time, can_cancel)
        @can_jump = can_jump
        @speed = speed
        @max_time = max_time
        @can_cancel = can_cancel
    end
    def self.read_from(stream)
        can_jump = stream.read_bool()
        speed = stream.read_double()
        max_time = stream.read_double()
        can_cancel = stream.read_bool()
        JumpState.new(can_jump, speed, max_time, can_cancel)
    end
    def write_to(stream)
        stream.write_bool(@can_jump)
        stream.write_double(@speed)
        stream.write_double(@max_time)
        stream.write_bool(@can_cancel)
    end
end

File: clients/python/model/jump_state.py
class JumpState:
    def __init__(self, can_jump, speed, max_time, can_cancel):
        self.can_jump = can_jump
        self.speed = speed
        self.max_time = max_time
        self.can_cancel = can_cancel
    @staticmethod
    def read_from(stream):
        can_jump = stream.read_bool()
        speed = stream.read_double()
        max_time = stream.read_double()
        can_cancel = stream.read_bool()
        return JumpState(can_jump, speed, max_time, can_cancel)
    def write_to(self, stream):
        stream.write_bool(self.can_jump)
        stream.write_double(self.speed)
        stream.write_double(self.max_time)
        stream.write_bool(self.can_cancel)
    def __repr__(self):
        return "JumpState(" + \
            repr(self.can_jump) + "," + \
            repr(self.speed) + "," + \
            repr(self.max_time) + "," + \
            repr(self.can_cancel) + \
            ")"
